src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxy.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Reflection
+{
+    /// <summary>
+    /// DispatchProxy provides a mechanism for the instantiation of proxy objects and handling of
+    /// their method dispatch.
+    /// </summary>
+    public abstract class DispatchProxy
+    {
+        protected DispatchProxy()
+        {
+#if NETSTANDARD2_0_OR_GREATER
```

Comment:
Is this the right directive to use?

Reply:
@ericstj was saying that we don't necessarily need this but if we keep it, we should pass the PNSE message that you already added to Resources.resx in to the exception.

I would just use `#if NETSTANDARD2_0` to make this explicit.